### PR TITLE
246 all device events revoke on start up

### LINKED ISSUE #246: All device events should revoke when a device starts up
sendDeviceDescription trigger is our device start up catch - will do it off of that

### PR DESCRIPTION
**Pull request by DanielSeehausen**
closes #246 

commits are clean - should be easiest to read through them!

- `sendDeviceDescription` (which is how we hook devices starting up) now deletes all device events
- backfilled a test I forgot to add in my last PR
- very minor refactoring

### CODE REVIEW

**Comment by DanielSeehausen** [comment]
fixed rebase issue

File: services/device/src/device/triggers/sendDeviceDescription.js [hunk 1]
```diff
@@ -13,17 +14,27 @@ const { logger } = require('envDependentLogger')
 
 const Device = require('../device')
 
+async function revokeDeviceEvents(deviceId) {
+  const revokedEventIds = await EventRevoker.revokeDeviceEvents(deviceId)
+  return Promise.all(revokedEventIds.map(async revokedId => {
```
[the remark this answers is not in the document]
Comment (DanielSeehausen) [reply services/device/src/device/triggers/sendDeviceDescription.js]: fixed later

File: services/device/test/trigger/createEvent.test.js [hunk 2]
```diff
@@ -139,10 +139,26 @@ describe('device createEvent trigger', () => {
...
+
+            const payload = await getValidPayloadForExistingDevice()
+            payload.name = 'doorClose'
+            payload.deviceId = device.id
+
+            await fireCreateEventEvent(payload)
+
+            expect(IotDataAPI.publishDeviceEvent).toHaveBeenCalledTimes(2)
+            expect(IotDataAPI.publishDeviceEvent)
+              .nthCalledWith(1, device.id, { revokedId: revocableEventA.id })
```
Comment (vdiezel) [inline services/device/test/trigger/createEvent.test.js]: :+1:

File: test/lib/models/EventRevoker.test.js [hunk 3]
```diff
@@ -46,6 +46,47 @@ describe('EventRevoker', () => {
...
+    })
+
+    it('returns ids of revoked events', async () => {
+      const emptyArray = await EventRevoker.revokeDeviceEvents(uuidv4())
+      expect(emptyArray).toEqual([])
+
+      const targetDevice = await Seeder.seedDevice()
+      const bystanderDevice = await Seeder.seedDevice()
+      const eventA = await Seeder.seedDeviceEvent(targetDevice, { name: 'This aggression will not stand, man...' })
+      const eventB = await Seeder.seedDeviceEvent(targetDevice, { name: 'Donnie was a good bowler, and a good man.' })
```
Comment (vdiezel) [inline test/lib/models/EventRevoker.test.js]: https://media.giphy.com/media/j7wBU7aHcKf7y/giphy.gif

**Comment by DanielSeehausen** [comment]
re-requested but only from rebase!

**Pull request merged**